### PR TITLE
tenv 1.10.0

### PR DESCRIPTION
**Pull request by kvendingoldo**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by kvendingoldo** [comment]
here the wrong formula (not from totuutils team) has been added https://github.com/Homebrew/homebrew-core/pull/170373/files

**Comment by chenrui333** [comment]
@kvendingoldo that was added by me, homebrew-core has to do the source build (always)

File: Formula/t/tenv.rb [hunk 1]
```diff
@@ -1,32 +1,80 @@
...
+            bin.install "terragrunt"
+            bin.install "tf"
+            bin.install "tofu"
+          end
+        end
+    end
   end
 
   test do
-    assert_match "1.6.2", shell_output("#{bin}/tenv tofu list-remote")
```
Comment (chenrui333) [inline Formula/t/tenv.rb:29]: no need to remove this line.

Comment (kvendingoldo) [reply Formula/t/tenv.rb:29]: why? I don't see anything similar in terraform Formula

Comment (chenrui333) [reply Formula/t/tenv.rb:29]: it is not trying to be the same

File: Formula/t/tenv.rb [hunk 2]
```diff
@@ -1,32 +1,80 @@
...
+            bin.install "tofu"
+          end
+        end
+    end
+    on_arm do
+        if Hardware::CPU.is_64_bit?
+          url "https://github.com/tofuutils/tenv/releases/download/v1.10.0/tenv_v1.10.0_Linux_arm64.tar.gz"
+          sha256 "1868e60775a1c1f41b502930ac921cf1cce2c94e0d67f0b12db6eadbb7c7a56e"
 
-    generate_completions_from_executable(bin/"tenv", "completion")
```
[the remark this answers is not in the document]
Comment (kvendingoldo) [reply Formula/t/tenv.rb:25]: returned back

File: Formula/t/tenv.rb (hunk 1 above)
Comment (kvendingoldo) [reply Formula/t/tenv.rb]: returned back

File: Formula/t/tenv.rb [hunk 3]
```diff
@@ -1,27 +1,79 @@
...
+      sha256 "1d13852b56b762353a28173400000dbf708ae6d5d6473353d2c26ab6b10e45de"
+
+      def install
+        bin.install "tenv"
+        bin.install "terraform"
+        bin.install "terragrunt"
+        bin.install "tf"
+        bin.install "tofu"
+      end
+    end
```
Comment (chenrui333) [inline Formula/t/tenv.rb:19]: we need to build from source, period

File: Formula/t/tenv.rb [hunk 4]
```diff
@@ -1,27 +1,79 @@
...
+    on_intel do
+      url "https://github.com/tofuutils/tenv/releases/download/v1.10.0/tenv_v1.10.0_Darwin_x86_64.tar.gz"
+      sha256 "1d13852b56b762353a28173400000dbf708ae6d5d6473353d2c26ab6b10e45de"
+
+      def install
+        bin.install "tenv"
+        bin.install "terraform"
+        bin.install "terragrunt"
+        bin.install "tf"
+        bin.install "tofu"
```
Comment (chenrui333) [inline Formula/t/tenv.rb:17]: also we already have each formula for the binaries, no need to install

from what I can say, the formula should only provide tenv binary

File: Formula/t/tenv.rb (hunk 3 above)
Comment (kvendingoldo) [reply Formula/t/tenv.rb:19]: we use goreleaser to do it. I'm not sure that we'll be able to keep two different brew Formulas. Can we delete Formula from homebrew-core then?

File: Formula/t/tenv.rb (hunk 4 above)
Comment (kvendingoldo) [reply Formula/t/tenv.rb:17]: terraform / terragrunt is a wrapper that are written by tofuutils, these are not real terraform or terragrunt

File: Formula/t/tenv.rb (hunk 3 above)
Comment (chenrui333) [reply Formula/t/tenv.rb:19]: it is separate build, you do yours, we do ours (and you can trust our build)

File: Formula/t/tenv.rb (hunk 4 above)
Comment (chenrui333) [reply Formula/t/tenv.rb:17]: okay, let me tweak your change a bit, like I did for the other one.

File: Formula/t/tenv.rb (hunk 3 above)
Comment (kvendingoldo) [reply Formula/t/tenv.rb:19]: yes, but Formula in this repo broken `tenv` utility for our users. It will be better to delete the `tenv` formula from here

Comment (chenrui333) [reply Formula/t/tenv.rb]: okay, I have installed all the binaries

File: Formula/t/tenv.rb [hunk 5]
```diff
@@ -18,11 +18,17 @@ class Tenv < Formula
...
+  conflicts_with "tfenv", because: "tfenv symlinks terraform binaries"
+  conflicts_with "tgenv", because: "tgenv symlinks terragrunt binaries"
+
   def install
     ldflags = "-s -w -X main.version=#{version}"
-    system "go", "build", *std_go_args(ldflags:), "./cmd/tenv"
-
-    generate_completions_from_executable(bin/"tenv", "completion")
+    %w[tenv terraform terragrunt tf tofu].each do |f|
+      system "go", "build", *std_go_args(ldflags:, output: bin/f), "./cmd/#{f}"
```
Comment (chenrui333) [inline Formula/t/tenv.rb:30]: some issue with completion, let's run the whole flow first

**Review by carlocab — APPROVED** [review]

**Pull request merged**